plugins/netskope/icon_netskope/util/api.py

```diff
@@ -0,0 +1,137 @@
+        update_file_hash_list_url = f"{self.api_url_v1}updateFileHashList/"
+        return self._call_api_v1("GET", update_file_hash_list_url, json_data={"token": self.api_key_v1, **data})
+
+    def get_all_url_list(self, params: dict) -> list:
+        get_all_url_list_url = f"{self.api_url_v2}policy/urllist/"
+        return self._call_api_v2("GET", get_all_url_list_url, params=params)
+
+    def create_a_new_url_list(self, params: dict) -> list:
+        create_a_new_url_list_url = f"{self.api_url_v2}policy/urllist/"
+        created_urllist = self._call_api_v2("POST", create_a_new_url_list_url, json_data=params)
+        return self.get_url_list_by_id(created_urllist[0].get("id"))
+
+    def upload_json_config(self, filename: str, content: bytes) -> dict:
+        upload_json_config_url = f"{self.api_url_v2}policy/urllist/file/"
+        return self._call_api_v2(
+            "POST", upload_json_config_url, files={"urllist": (filename, base64.b64decode(content))}
+        )
+
+    def get_url_list_by_id(self, identifier: int) -> dict:
+        get_url_list_by_id_url = f"{self.api_url_v2}policy/urllist/{identifier}"
+        return self._call_api_v2("GET", get_url_list_by_id_url)
+
+    def replace_url_list_by_id(self, identifier: int, data: dict) -> dict:
+        replace_url_list_by_id_url = f"{self.api_url_v2}policy/urllist/{identifier}"
+        self._call_api_v2("PUT", replace_url_list_by_id_url, json_data=data)
+        return self.get_url_list_by_id(identifier)
+
+    def delete_url_list_by_id(self, identifier: int) -> dict:
+        delete_url_list_by_id_url = f"{self.api_url_v2}policy/urllist/{identifier}"
+        self._call_api_v2("DELETE", delete_url_list_by_id_url)
+        return self.get_url_list_by_id(identifier)
+
+    def patch_url_list_by_id(self, identifier: int, action: str, data: dict) -> dict:
+        patch_url_list_by_id_url = f"{self.api_url_v2}policy/urllist/{identifier}/{action}"
+        self._call_api_v2("PATCH", patch_url_list_by_id_url, json_data=data)
+        return self.get_url_list_by_id(identifier)
+
+    def apply_pending_url_list_changes(self) -> list:
+        apply_pending_url_list_changes_url = f"{self.api_url_v2}policy/urllist/deploy/"
+        deployed_lists = self.get_all_url_list({"pending": 1})
+        self._call_api_v2("POST", apply_pending_url_list_changes_url)
+        return deployed_lists
+
+    def get_single_user_confidence_index(self, data: dict) -> dict:
+        get_single_user_uci_url = f"{self.api_url_v2}ubadatasvc/user/uci"
+        return self._call_api_v2("POST", get_single_user_uci_url, json_data=data)
+
+    def test_api(self) -> dict:
+        GET_TEST_URL = f"{self.api_url_v1}updateFileHashList/"
+        self.test_connection = True
+        self._call_api_v1("GET", GET_TEST_URL, json_data={"token": self.api_key_v1})
+        return self.get_all_url_list({"pending": 1})
+
+    def _rate_limiting(max_tries: int) -> dict:
```

Comment:
Maybe add a docstring documenting that decorator. In general we should move this backoff to some common util lib as everyone is implementing their own backoff function.

Reply:
I will update that with docstring